src/Microsoft.OData.Core/Evaluation/ODataMetadataContext.cs

```diff
@@ -43,8 +43,9 @@ internal interface IODataMetadataContext
         /// </summary>
         /// <param name="resourceState">Resource state to use as reference for information needed by the builder.</param>
         /// <param name="useKeyAsSegment">true if keys should go in separate segments in auto-generated URIs, false if they should go in parentheses.</param>
+        /// <param name="isDelta">true if the payload being read is a delta payload</param>
         /// <returns>An entity metadata builder.</returns>
-        ODataResourceMetadataBuilder GetResourceMetadataBuilderForReader(IODataJsonLightReaderResourceState resourceState, bool useKeyAsSegment);
+        ODataResourceMetadataBuilder GetResourceMetadataBuilderForReader(IODataJsonLightReaderResourceState resourceState, bool useKeyAsSegment, bool isDelta = false);

```

Comment:
>bool isDelta = false [](start = 145, length = 20)

maybe it's better not to use optional parameter? But, it's up to you.

Reply:
Yes, I was just being lazy. Fixed.

---
In reply to: [256641618](https://github.com/OData/odata.net/pull/1404#discussion_r256641618) [](ancestors = 256641618)